launcher/ui/widgets/ModFilterWidget.cpp

```diff
@@ -16,6 +18,24 @@ ModFilterWidget::ModFilterWidget(Version def, QWidget* parent)
 
     m_filter->versions.push_front(def);
 
+    m_version_list = APPLICATION->metadataIndex()->get("net.minecraft");
+    if (!m_version_list->isLoaded()) {
+        QEventLoop load_version_list_loop;
+
+        auto task = m_version_list->getLoadTask();
+
+        connect(task.get(), &Task::failed, [this]{
+            ui->majorVersionButton->setText(tr("Major version match (failed to get version index)"));
+            disableVersionButton(VersionButtonID::Major);
+        });
+        connect(task.get(), &Task::finished, &load_version_list_loop, &QEventLoop::quit);
+
+        if (!task->isRunning())
+            task->start();
+
+        load_version_list_loop.exec();
```

Comment:
What's the event loop needed for?

Reply:
To wait for the version list to load before showing the widget, so that if an error occurs we can disable the `majorVersionButton` before showing it to the user, and if the list loaded successfully, the filter option works at all.

Reply:
> To wait for the version list to load before showing the widget, so that if an error occurs we can disable the `majorVersionButton` before showing it to the user, and if the list loaded successfully, the filter option works at all.

Why can't this be done by just hiding the widget first and connecting to a slot that shows it?

Reply:
hiding the widget and only showing it later seems more like a bug. ideally, when the dialog opens, you'd have everything loaded too, lazy-loading this kind of stuff doesn't look right.

regarding having it in the class constructor, even though it shouldn't cause any issues, we could just have it be done in a factory method, the main point is to have it available by the time the user sees the UI.

Reply:
So is there also no way to show the window with a slot after it's loaded?

Reply:
there can be a way to do that, but why? i don't see much problem in using a QEventLoop here, it's simpler than doing whatever you'd need to to delay the window's show() until the list is loaded, and from testing, the delay caused by it isn't even noticeable for me..

Reply:
It feels wrong to me but maybe my feeling is the thing that's wrong. I don't really know whether it's a bad approach it was just my gut feeling. If you're sure this is fine and there's no caveats to using a local event loop here that would cause any problem then I'll approve it.

Reply:
i've moved the thing to a factory method, and added a QTimer on it to be extra extra sure. this should make it somewhat similar to what we do on other parts of the code base, so shouldn't be any problems there.